### PR TITLE
Add `create_user` and `create_workspace` management commands.

### PR DESCRIPTION
**Pull request by bloodearnest**
- Add create_user management command
- Add create_workspace management command

These, along with #4075, allow for automated set up of a job-server instance for integration testing with airlock.

### CODE REVIEW

File: jobserver/management/commands/create_user.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+        if options["name"]:
+            update_fields.append("fullname")
+
+        user, created = get_or_create_user(
+            username,
+            email=options["email"] or f"{username}@example.com",
+            fullname=options["name"] or username,
+            update_fields=update_fields,
+        )
+        print(f"User {username} {'created' if created else 'updated'}")
```
Comment (rebkwok) [inline jobserver/management/commands/create_user.py:50]: As recommended in django docs:
```suggestion
        self.stdout.write(f"User {username} {'created' if created else 'updated'}")
```

File: jobserver/management/commands/create_user.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        if options["output_checker"]:
+            roles.append(OutputChecker)
+        if options["core_developer"]:
+            roles.append(CoreDeveloper)
+
+        updated = False
+        for role in roles:
+            if role not in user.roles:
+                updated = True
+                print(f"Added {role.__name__} global role to user {user.username}")
```
Comment (rebkwok) [inline jobserver/management/commands/create_user.py:62]:
```suggestion
                self.stdout.write(f"Added {role.__name__} global role to user {user.username}")
```

File: jobserver/management/commands/create_user.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+                user.roles.append(role)
+            else:
+                print(f"User {user.username} already has role {role.__name__}")
```
Comment (rebkwok) [inline jobserver/management/commands/create_user.py:65]:
```suggestion
                self.stdout.write(f"User {user.username} already has role {role.__name__}")
```

File: jobserver/management/commands/create_workspace.py [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+            "name": project_slug,
+            "created_by": user,
+            "updated_by": user,
+        }
+
+        project, created = Project.objects.update_or_create(
+            slug=project_slug, defaults=project_values
+        )
+
+        print(f"{'Created' if created else 'Updated'} project {project.slug}")
```
Comment (rebkwok) [inline jobserver/management/commands/create_workspace.py:55]:
```suggestion
        self.stdout.write(f"{'Created' if created else 'Updated'} project {project.slug}")
```

File: jobserver/management/commands/create_workspace.py [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+            "created_by": user,
+            "updated_by": user,
+            "repo": repo,
+            "project": project,
+        }
+
+        workspace, created = Workspace.objects.update_or_create(
+            name=options["workspace"], defaults=workspace_values
+        )
+        print(f"{'Created' if created else 'Updated'} workspace {workspace.name}")
```
Comment (rebkwok) [inline jobserver/management/commands/create_workspace.py:72]:
```suggestion
        self.stdout.write(f"{'Created' if created else 'Updated'} workspace {workspace.name}")
```

File: jobserver/management/commands/create_workspace.py [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+
+        if user not in project.members.all():
+            project_members.add(
+                project=project,
+                user=user,
+                roles=[ProjectDeveloper, ProjectCollaborator],
+                by=user,
+            )
+            print(
```
Comment (rebkwok) [inline jobserver/management/commands/create_workspace.py:81]:
```suggestion
            self.stdout.write(
```

**Review by rebkwok — APPROVED** [review]
LGTM, but should probably have a REX team review as well

**Review by lucyb — APPROVED** [review]

**Pull request merged**